ArduCopter/GCS_Mavlink.cpp

```diff
@@ -583,6 +583,49 @@ MAV_RESULT GCS_MAVLINK_Copter::handle_command_int_packet(const mavlink_command_i
 #endif
         return MAV_RESULT_UNSUPPORTED;
 
+      case MAV_CMD_DO_REPOSITION:
+            {
+                const bool change_modes = ((int32_t)packet.param2 & MAV_DO_REPOSITION_FLAGS_CHANGE_MODE) == MAV_DO_REPOSITION_FLAGS_CHANGE_MODE;
+                if (!copter.flightmode->in_guided_mode() && !change_modes) {
+                    return MAV_RESULT_FAILED;
+                }
+
+                Location requested_position {};
+                requested_position.lat = packet.x;
+                requested_position.lng = packet.y;
+
+                // check the floating representation for overflow of altitude
+                if (fabsf(packet.z * 100.0f) >= 0x7fffff) {
+                    return MAV_RESULT_FAILED;
+                }
+                requested_position.alt = (int32_t)(packet.z * 100.0f);
+
+                // load option flags
+                if (packet.frame == MAV_FRAME_GLOBAL_RELATIVE_ALT_INT) {
+                    requested_position.relative_alt = 1;
+                } else if (packet.frame == MAV_FRAME_GLOBAL_TERRAIN_ALT_INT) {
+                    requested_position.terrain_alt = 1;
+                } else if (packet.frame != MAV_FRAME_GLOBAL_INT &&
+                         packet.frame != MAV_FRAME_GLOBAL) {
+                    // not a supported frame
+                    return MAV_RESULT_FAILED;
+                }
+
+                if (requested_position.sanitize(copter.current_loc)) {
+                    // if the location wasn't already sane don't load it
+                    return MAV_RESULT_FAILED; // failed as the location is not valid
```

Comment:
`MAV_RESULT_DENIED`?

Reply:
I had the thought, but I was trying to keep the failure cases between plane and copter similar. If you think we should change plane in these cases then I'm happy to change it here, but I do think those should look the same.